html_css/v2/main-course/intermediate_css/frameworks-preprocessors.md

```diff
@@ -0,0 +1,91 @@
+
+For the reasons above, **we recommend that you continue to focus on the fundamentals of CSS as you continue through the curriculum.** This lesson is merely designed to teach you _about_ frameworks and their role.
+
+### Preprocessors Overview
+
+Preprocessors (aka precompilers) are languages that help you write CSS more easily. They  eliminate code repetition and provide all sorts of time-saving and code-saving features. Some of the standard preprocessors in use are [SASS](https://sass-lang.com/), [LESS](https://lesscss.org/)and [Stylus](https://stylus-lang.com/).
+
+Note: If you've been using Rails, you'll be familiar with SASS already since it's baked in.  If you're not using Rails, you may need to add some additional steps to your workflow to get it set up, but it's usually worth it.
+
+Some of the useful features of preprocessors include:
+
+1. Programming in CSS but with more options available to you.
+2. Including variables in your CSS (not an option in vanilla CSS).
+3. Including loops in your CSS (again not an option in vanilla CSS).
+4. Including if/else statement in your CSS (see the pattern here?).
+5. Allowing you to create reusable code.
+6. And a lot more features, depending on which preprocessor you use.
+
+As you can see, preprocessors are extremely helpful tools; however, there are some potential disadvantages to using them. See the reading below for one list of disadvantages.
+
+### Assignment
+
+<div class="lesson-content__panel" markdown="1">
+1. Read [this brief overview of frameworks](https://medium.com/html-all-the-things/what-is-a-css-framework-f758ef0b1a11).
+2. Check out [this article](https://www.smashingmagazine.com/2018/11/css-frameworks-css-grid/) on the pros and cons of using a framework vs. CSS grid.
+3. Read [this brief article](https://blog.zipboard.co/do-i-need-to-start-using-css-preprocessors-a72b525c180a) that takes a look at the three preprocessors mentioned above, and why they are useful.
+4. Read [this extremely brief article](https://medium.com/@LiliOuakninFelsen/what-is-a-css-pre-processor-why-should-you-use-one-ac3d4cd4b2c2) on why you should use a preprocessor.
+5. For balance, read [this list of the disadvantages of using a preprocessor](https://adamsilver.io/blog/the-disadvantages-of-css-preprocessors/).
+</div>
+
+### Additional Resources
+
+* [Bootstrap documentation](https://bootstrapdocs.com/v3.0.3/docs/getting-started/)
+* [Foundation documentation](https://get.foundation/sites/docs/)
+* [Tailwind documentation](https://tailwindcss.com/docs)
+* [Bulma documentation](https://bulma.io/documentation/)
+* [SASS documentation](https://sass-lang.com/documentation)
+* [LESS documentation](https://lesscss.org/)
+* [Stylus documentation](https://stylus-lang.com/)
```

Comment:
Having every piece of documentation for each framework/preprocessor seems to be a bit much. The goal of this lesson should be a conceptual overview, the additional resources should also reflect that. I don't think we want to suggest any of these for use.

Reply:
OK, I can just put documentation for like Bootstrap, Foundation and SASS maybe. Sound good?

Reply:
Bootstrap, Tailwind and SASS imo.  Foundation isn't cool anymore